### PR TITLE
Main release workflow

### LINKED ISSUE #16: Configure release workflow for testing, and tagging new SDK verisons
Packer has an automated workflow for running acceptance testing before cutting a release. The workflow will run tests, update the change log, create the version release tag, and finally publish the bins. The packer-plugin-sdk repo should have a similar workflow for testing, updating the change log, and tagging the latest code base with the appropriate release version; no bin publishing necessary as this is meant to be a Go module. 

See [hashicorp/terraform-plugin-sdk](https://github.com/hashicorp/terraform-plugin-sdk/blob/master/.circleci/config.yml) for an example

### PR DESCRIPTION
**Pull request by nywilken**
This change configures a simple CircleCI workflow for releasing the Packer Plugin SDK from main. The base for this change was taken from terraform-plugin-sdk release model documented [here](https://github.com/hashicorp/engineering-docs/blob/7f298fbc2ddc103bc5369aa4ba0c5188763e95ff/terraform/releasing_sdk.md)

Closes #16 

 The workflow is as follows:

Upon pushing to master a trigger-release job will be executed, which will wait for approval from a maintainer.
Once approved the trigger-release job will run the script `scripts/release/release.sh`, which will execute the following steps:
 

1. Update the CHANGELOG with the release date
2. Update the SDK version in `version/version.go`
3. Commit the changes to the main branch
4. Create a tag for the version (vX.Y.Z)
5. Push the release commits and tag to main branch

### TODO
- [x]  Configure GPG signing for commit and tag creation
- [x] Test the process :smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,66 @@ Migrating to the standalone SDK v1 is covered on the [Plugin SDK section](https:
...
+where `x.y.z` is the SDK version you intend to release.
+
+
+Underneath the `# x.y.z (Upcoming)` heading, please write human-readable entries describing the changes made in this version.
+
+### Triggering a release
+
+All commits to `main` trigger the `release` workflow on CircleCI, but manual approval is needed to proceed with the workflow.
+
+#### Find all `master` workflows
```
Comment (azr) [inline README.md:76]: Q: Should this be main here ?

```suggestion
#### Find all `main` workflows
```

**Review by azr — APPROVED** [review]
Nice one LGTM, I just wondered if we could sort of avoid having to maintain the generator but I also like the simplicity of just using sed.

Generators I found:
* https://github.com/github-changelog-generator/github-changelog-generator
* https://github.com/hashicorp/go-changelog
* https://github.com/twisted/towncrier
* https://github.com/paultyng/changelog-gen

File: README.md (hunk 1 above)
Comment (nywilken) [reply README.md:76]: Yes, thank you.

**Comment by nywilken** [comment]
> Nice one LGTM, I just wondered if we could sort of avoid having to maintain the generator but I also like the simplicity of just using sed.
> 
> Generators I found:
> 
>     * https://github.com/github-changelog-generator/github-changelog-generator
> 
>     * https://github.com/hashicorp/go-changelog
> 
>     * https://github.com/twisted/towncrier
> 
>     * https://github.com/paultyng/changelog-gen

Using an CHANGELOG generator is a great idea. It looks like we need to add some PR templates to get them working which is easy to add. Do you think it should be part of this PR?

**Comment by nywilken** [comment]
@hashicorp/packer merging this to get the release flow working on master. I would like to revisit the CHANGELOG generation in a separate PR.

**Pull request merged**